Add the `--non-interactive` flag when installing type stubs via `mypy`

jsf9k opened the pull request:
## 🗣 Description ##

This pull reques
- Adds the `--non-interactive` flag when installing type stubs via `mypy`
- Moves the `mypy` command toward the end of the `setup-env` script

## 💭 Motivation and context ##

- This flag forces `mypy` to hide the errors about missing stubs and instead simply install stubs without asking for confirmation.  It also does not return an error code, which it does without this flag _even if you opt to let it install the missing type stubs_.
- Moving the `mypy` command toward the end of the `setup-env` script ensures that, should the command fail, all the other good stuff in the script will still happen.

## 🧪 Testing ##

All automated tests pass.  These changes were tested as part of cisagov/cyhy-config#5.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

jsf9k committed:
Add the --non-interactive flag when installing type stubs via mypy

This flag forces mypy to hide the errors about missing stubs and
instead simply install stubs without asking for confirmation.  It also
does not return an error code, which it does without this flag even if
you opt to let it install the missing type stubs.

dav3r reviewed (approved):
I'm on board with this change. 👍

jsf9k committed:
Move mypy command toward the end of the script

This allows all the other good things to happen even if the mypy
command fails.
